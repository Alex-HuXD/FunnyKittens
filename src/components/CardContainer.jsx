import React from 'react';
import Card from './Card';
import TellMe from './TellMe';

function CardContainer({ cats }) {
  let toggle = false;
  cats.length === 1 ? (toggle = true) : (toggle = false);

  return (
    <div
      className="card-container pa3"
      style={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'center' }}
    >
      {cats.map((cat, i) => {
        return (
          <Card
            key={i}
            name={cats[i].name}
            id={cats[i].id}
            email={cats[i].emai}
          />
        );
      })}

      {toggle === false ? null : <TellMe />}
    </div>
  );
}

export default CardContainer;
