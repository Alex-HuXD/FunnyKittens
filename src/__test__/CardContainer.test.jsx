import { render, screen, fireEvent } from '@testing-library/react';

import CardContainer from '../components/CardContainer';

const mockedCardsSingle = [{ name: 'testName', email: 'testEmail', id: 1 }];

const mockedCardsMutiple = [
  { name: 'testName', email: 'testEmail', id: 1 },
  { name: 'testName1', email: 'testEmail1', id: 2 },
];

describe('CardContainer', () => {
  screen.debug();

  it('should display same as snapshot && if style change is intentional then snapshot must be updated', () => {
    render(<CardContainer cats={mockedCardsSingle} />);
    expect(document.body.innerHTML).toMatchSnapshot();
  });

  it('should render tellMe component when only one card is displayed', () => {
    render(<CardContainer cats={mockedCardsSingle} />);
    const tellMeComponent = screen.getByRole('button', { name: /speak/i });
    expect(tellMeComponent).toBeInTheDocument();
  });

  it('should not render tellMe component when more than one card is displayed', () => {
    render(<CardContainer cats={mockedCardsMutiple} />);
    const tellMeComponent = screen.queryByRole('button', { name: /speak/i });
    expect(tellMeComponent).toBeFalsy();
  });
});
