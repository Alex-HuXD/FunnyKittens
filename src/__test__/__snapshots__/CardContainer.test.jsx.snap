// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should display same as snapshot && if style change is intentional then snapshot must be updated 1`] = `"<div><div class=\\"card-container pa3\\" style=\\"display: flex; flex-wrap: wrap; justify-content: center;\\"><div class=\\" bg-light-green white-90 dib br3 pa2 ma3 shadow-5 grow\\"><img src=\\"https://robohash.org/1?set=set4&amp;size=250x250\\" alt=\\"card\\"><div><p>testName</p><p></p></div></div><div class=\\" tell-me tc dib br3 bg-light-green\\"><img class=\\"dib f3\\" src=\\"https://www.pinclipart.com/picdir/big/562-5622317_crystal-ball-emoji-png-emoji-crystal-ball-png.png\\" alt=\\"ball\\" style=\\"height: 200px; width: auto; padding: 15px;\\"><div class=\\"btn-container\\"><button class=\\"shadow-5\\">click me</button><button>Speak!!!</button></div><p></p></div></div></div>"`;
